### ISSUE TITLE
URL of the final wikipage

### ISSUE THREAD

**Issue by lawrencenull** [spencermountain/wtf_wikipedia#260]
Was going through the API, is there a current way to generate a link to the wiki URL that the info is being pulled from.

**Comment by spencermountain** [comment]
hey lawrence, no there isn't a clean one. Maybe there should be.
right now i'd just do something like this:
```js
let title=doc.title().replace(/ /g,'_')
title=encodeURIComponent(title)
let url=`${lang}.wikipedia.org/wiki/${title}}`
```
we need to store the language for the upcoming image url refactor, so maybe we should add this in after that.
cheers

**Comment by niebert** [comment]
When you pull from an other wiki domain - e.g. [Wikiversity](https://en.wikiversity.org)  
* https://en.wikiversity.org/wiki/Swarm_intelligence
* https://en.wikiversity.org/wiki/Water
* https://en.wikiversity.org/wiki/Risk_Management
then the title replacement is the same, but you need also a domain replace in the generate url
the url should be set like this
```javascript
/* FIle: /src/_fetch/_title2url
   function that converts a title from domain Wikipedia, Wikiversity, .... and a specific
   language "lang" and a domain "wikipedia", "wikiversity" into an URL that displays the source.
*/

const title2url = function( title, lang, domain ) {
   // replace blank to underscore 
   let title=doc.title().replace(/ /g,'_')
   title = encodeURIComponent(title);
   let url=`${lang}.${domain}.org/wiki/${title}}`
  return url;
};

module.exports =title2url;
```
Maybe Spencer could provide the function as file in `/src/_fetch/_title2url.js` for further use or integrate `title2url` into the API. See also [Github-Wiki](https://github.com/spencermountain/wtf_wikipedia/wiki)

cheers,

**Comment by spencermountain** [comment]
added `.language()` and `.url()` methods, will release as `8.1.0`
cheers

**Comment by skipjack** [comment]
Can you ping this thread once 497e075d1 makes it into `master` and/or when the new release is cut?

**Comment by spencermountain** [comment]
sure

**Comment by spencermountain** [comment]
released as `8.1.0`
cheers

**Comment by skipjack** [comment]
Works like a charm, thanks!

**Comment by spencermountain** [comment]
ah, good eye.
fixed on dev, will add to next release.

**Comment by spencermountain** [comment]
fixed in `8.2.1`
